src/main/java/com/google/cloud/tools/Action.java

```diff
@@ -0,0 +1,144 @@
+   *
+   * @return {@code true} if execution was successful, {@code false} if it was unsuccessful but no
+   * exception was thrown.
+   * @throws GCloudErrorException if any error invoking or running gcloud occurred.
+   */
+  public boolean execute() throws GCloudErrorException {
+    return processCaller.call();
+  }
+
+  /**
+   * Flags accepted by an {@link Action}.
+   *
+   * <p>Every {@link Action} implementation has an {@code acceptedFlags} static array with its
+   * accepted flags. This is a utility method that transforms that array into an iterable
+   * {@link Set}.
+   */
+  @VisibleForTesting
+  public Set<Option> getAcceptedFlags() {
+    return new HashSet<>(Arrays.asList(acceptedFlags));
+  }
+
+  @VisibleForTesting
+  public ProcessCaller getProcessCaller() {
+    return processCaller;
+  }
+
+  /**
+   * Checks if {@code flags} contains flags that are not accepted by the {@link Action} and if
+   * any flag's value is invalid.
+   *
+   * @throws InvalidFlagException if client-provided flags contain flags not accepted by the
+   * {@link Action}.
+   */
+  protected void checkFlags(Map<Option, String> clientProvidedFlags) {
+    Set<Option> acceptedFlags = getAcceptedFlags();
+
+    for (Entry<Option, String> clientFlag : clientProvidedFlags.entrySet()) {
+      // Check for unrecognised flags.
+      if (!acceptedFlags.contains(clientFlag.getKey())) {
+        throw new InvalidFlagException(String.format(
+            "The %s flag is not recognised by the %s command.",
+            clientFlag.getKey().getLongForm(), this.getClass().getName()));
+      }
+
+      // Check for invalid values.
+      String flagValue = clientFlag.getValue();
+      if (clientFlag.getKey().getType().equals(Type.BOOLEAN)) {
+        // If boolean, either there is no value, or the value has to be "true" or "false".
+        if (!Strings.isNullOrEmpty(flagValue)) {
+          if (!flagValue.toLowerCase().equals("true") && !flagValue.toLowerCase().equals("false")) {
+            throw new InvalidFlagException(String.format(
+                "The value %s for flag %s is invalid.",
+                flagValue, clientFlag.getKey().getLongForm()));
+          }
+        }
+      } else if (clientFlag.getKey().getType().equals(Type.INTEGER)) {
+        // If integer, flag must have a value and be int parsable.
+        if (!Strings.isNullOrEmpty(flagValue)) {
+          try {
+            Integer.parseInt(flagValue);
+          } catch (NumberFormatException nfe) {
+            throw new InvalidFlagException(String.format(
+                "The value %s for flag %s is invalid.",
+                flagValue, clientFlag.getKey().getLongForm()));
+          }
+        } else {
+          throw new InvalidFlagException(String.format(
+              "Flag %s must have a value.", clientFlag.getKey().getLongForm()));
+        }
+      }
+    }
+  }
+
+  /**
+   * Sets a new Cloud SDK location, in case the is isn't installed in the default user's home
```

Comment:
This is called from library clients, to specify different locations when Cloud SDK isn't installed in the default directory.